### PR TITLE
Map version hash to human readable string

### PR DESCRIPTION
**Pull request by kylestein-arm**
Signed-off-by: Kyle Stein <kyle.stein@arm.com>

### CODE REVIEW

**Comment by kylestein-arm** [comment]
still needs testing.  do not merge at this time

File: files/edge-core/scripts/edge-core-bootloader.sh [hunk 1]
```diff
@@ -132,6 +132,11 @@ if [ -e "${UPGRADE_WORKDIR}" ]; then
         check_error "pre-refresh" $? 1
         rm pre-refresh.sh
     fi
+    if [ -f map-version.json ]; then
+        echo "Adding new version mappings"
+        jq -s '.[0] * .[1]' ${VERSION_MAP} map-version.json > ${VERSION_MAP}.new
```
Comment (costanic) [inline files/edge-core/scripts/edge-core-bootloader.sh:137]: can you explain this what's happening here?  I would have expected this to be a simple file copy.

Comment (kylestein-arm) [reply files/edge-core/scripts/edge-core-bootloader.sh:137]: That jq command does a json dictionary merge.  So if the incoming json file is  {"abcdef" : "1.10.0"} and the existing map is {"12345" : "1.9.1", "67890", "1.9.0"}, then the resultant mapping will be {"abcdef" : "1.10.0", "12345" : "1.9.1", "67890", "1.9.0"}

**Review by johnmichaelwelsh — APPROVED** [review]
Looks good.

File: files/edge-core/scripts/edge-core-bootloader.sh (hunk 1 above)
Comment (mray190) [reply files/edge-core/scripts/edge-core-bootloader.sh]: Will copy rather than merge per discussion in standup/post-standup

**Review by mray190 — APPROVED** [review]

**Review by costanic — APPROVED** [review]

**Pull request merged**